### PR TITLE
disable mixed precision inference for quantized IRs

### PR DESCRIPTION
**Pull request by pavel-esir**
### Details:
 - Accidentally quantized models were marked by mixed precision inference transformations, because of that redundant Reorders were inserted around MVNs/NormalizeL2/etc. and this caused significant perf degradation on GPU. Need to disable mixed precision marking for Quantized IRs, on GPU they should run as is, without mixed precision inference.
 
image

### Tickets:
 - xxx-105969, xxx-105064

### CODE REVIEW

File: src/common/transformations/src/transformations/convert_precision.cpp [hunk 1]
```diff
@@ -312,8 +312,15 @@ bool ov::pass::ConvertPrecision::run_on_model(const std::shared_ptr<ngraph::Func
         return false;
 
     bool has_fp16_compression = m_precisions.count(element::f32) > 0 && m_precisions[element::f32] == element::f16;
+    bool has_quantize_nodes = false;
+    for (const auto& op : f->get_ops()) {
+        if (std::dynamic_pointer_cast<opset8::FakeQuantize>(op)) {
```
Comment (rkazants) [inline src/common/transformations/src/transformations/convert_precision.cpp:317]:
```suggestion
        if (as_typ_ptr<opset8::FakeQuantize>(op)) {
```

**Review by rkazants — APPROVED** [review]

**Comment by rkazants** [comment]
@KodiaqQ, @AlexKoff88, please review it before merge

**Comment by AlexKoff88** [comment]
cc'ed @vladimir-paramuzov.

File: src/common/transformations/src/transformations/convert_precision.cpp (hunk 1 above)
Comment (pavel-esir) [reply src/common/transformations/src/transformations/convert_precision.cpp:318]: these lines have been removed, but applied change dynamic_pointer_cast -> as_type_ptr in other modified places

**Comment by pavel-esir** [comment]
Closing, since solution here has not general. Implemented better solution with selective turning off marking here https://github.com/openvinotoolkit/openvino/pull/16785

**Pull request closed without merging**